test/parallel/test-worker-message-port-transfer-native.js

```diff
@@ -0,0 +1,31 @@
+// Flags: --expose-internals
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { MessageChannel } = require('worker_threads');
+const { internalBinding } = require('internal/test/binding');
+
+// Test that passing native objects and functions to .postMessage() throws
+// DataCloneError exceptions.
+
+{
+  const { port1, port2 } = new MessageChannel();
+  port2.once('message', common.mustNotCall());
+
+  assert.throws(() => {
+    port1.postMessage(function foo() {});
+  }, /^DataCloneError: function foo\(\) \{\} could not be cloned\.$/);
```

Comment:
```suggestion
  },
  {
    name: 'DataCloneError',
    message: /function foo\(\) \{\} could not be cloned\.$/
  });
```

Reply:
@joyeecheung Done, for both uses of `assert.throw()` :+1: